cli/resources/org/partiql/cli/syntax/PartiQL.nanorc

```diff
@@ -0,0 +1,34 @@
+syntax "PartiQL" "\.pql$"
```

Comment:
I'm not too familiar w/ JLine's syntax highlighter requirements but is it restricted to just `nanorc` files? I know @jpschorr had worked on some syntax highlighters for Ace, Python Pygments, Ruby Rouge, and Sublime Text (example integration in this recent Rust PR: https://github.com/partiql/partiql-lang-rust/pull/136). It would be nice to use one of those syntax highlighters or to use his generator to create a `nanorc` syntax highlighter to keep a consistent + complete syntax highlighting experience.

Reply:
An added benefit would be Ion syntax highlighting within PartiQL queries.

Reply:
^^ agreed. We can actually implement the highlighter however we want. Nanorc was chosen because this is built into JLine3, but you can actually implement your own highlighting. What I've typically done (which is what Trino does) is run the input through the lexer AND parser. The lexer will get you token types, and then you can even use the parser to highlight everything red if there's a parse error. It requires having a catch-all "unrecognized" rule

Reply:
New revision uses the SqlLexer and SqlParser instead of nanorc. Better experience overrall. See the new video in the description.

Reply:
^^ ask and you shall receive haha. This is awesome, nice work!